src/scroll-timeline-css-parser.js

```diff
@@ -135,16 +135,36 @@ export class StyleParser {
     return null;
   }
 
-  getViewTimelineOptions(timelineName) {
-    // TODO: Take into account the scoping of the ViewTimelines
-    // https://github.com/w3c/csswg-drafts/issues/7047
+  findPreviousSiblingOrAncestorMatchingSelector(target, selector) {
+    // Target self is eligible, use that
+    if (target.matches(selector)) return target;
+
+    // Walk previous siblings
+    let prevSibling = target.previousElementSibling;
+    while (prevSibling) {
+      if (prevSibling.matches(selector)) {
+        return prevSibling;
+      }
+      prevSibling = prevSibling.previousElementSibling;
+    }
+
+    // Move up to parent
+    if (target.parentElement) {
+      return this.findPreviousSiblingOrAncestorMatchingSelector(target.parentElement, selector);
```

Comment:
Nit: iteration is more efficient than recursion.
```
let candidate = target;
while (candidate) {
   if (candidate.matches(selector)) 
     return candidate;
   candidate = candidate.previousElementSibling || candidate.parentElement;
}
```

Reply:
Clever! Addressed in 31b3d9585daa88455b69d170a16b5e7d68aad6d9